x-pack/legacy/plugins/uptime/public/hooks/use_index_pattern.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, Dispatch } from 'react';
+import { HttpHandler } from 'kibana/public';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+
+const getIndexPattern = async (fetch: HttpHandler): Promise<any> => {
+  return await fetch('/api/uptime/index_pattern', { method: 'GET' });
+};
+
+export const useIndexPattern = <T>(setIndexPattern: Dispatch<T>) => {
+  const core = useKibana();
+  useEffect(() => {
+    const fetch = core.services.http?.fetch;
+    if (!fetch) throw new Error('Http core services are not defined');
+    getIndexPattern(fetch)
+      .then(indexPattern => setIndexPattern(indexPattern))
+      .catch(e => {
+        throw e;
```

Comment:
Why are we using promises explicitly here instead of async/await? The default behavior of `await` is to re-throw right?

Reply:
The primary reason I used it was probably expediency while trying to get this working and never going back and changing it. An annoying thing about `useEffect` is it can't handle an `async` function, so you need to use [a pattern like this one](https://github.com/facebook/react/issues/14326#issuecomment-441680293) to do asynchronous stuff inside an effect.

I will rewrite it to look like:
```typescript
useEffect(() => {
  const fetch = core.services.http?.fetch;
  async function getIndexPattern() {
    if (!fetch) throw new Error('Http core services are not defined');
    setIndexPattern(await fetch('/api/uptime/index_pattern', { method: 'GET' }));
  }
  getIndexPattern();
}, []);
```

Reply:
See https://github.com/elastic/kibana/pull/53726/commits/6a00da3426b967c7c4367b7b57e7d373decfb8a0.